iminuit/_libiminuit.pyx

```diff
@@ -611,6 +611,42 @@ cdef class Minuit:
         matrix = self.matrix(correlation=correlation, skip_fixed=skip_fixed)
         return np.array(matrix, dtype=np.float64)
 
+    def np_values(self):
+        """Parameter values in numpy array format."""
+        import numpy as np
+        return np.asarray(self.args, dtype=np.double)
+
+    def np_errors(self):
+        """Parameter errors in numpy array format."""
+        import numpy as np
+        a = np.empty(len(self.parameters), dtype=np.double)
+        for i, k in enumerate(self.parameters):
+            a[i] = self.errors[k]
+        return a
+
+    def np_merrors(self):
+        """Parameter errors in numpy array format."""
+        import numpy as np
+        # array format follows matplotlib conventions, see pyplot.errorbar
+        a = np.empty((2, len(self.parameters)), dtype=np.double)
+        for i, k in enumerate(self.parameters):
+            a[0, i] = -self.merrors[(k, -1.0)]
+            a[1, i] = self.merrors[(k, 1.0)]
+        return a
+
+    def np_covariance(self):
+        """Covariance matrix in numpy array format.
+
+        Note that a ``numpy.ndarray`` is returned, not a ``numpy.matrix``
+        """
+        import numpy as np
+        n = len(self.parameters)
+        a = np.empty((n, n), dtype=np.double)
```

Comment:
Why not put `return self.np_matrix(skip_fixed=False)` here?
It's less code for us to maintain, and guarantees that `np.covariance` is consistent with `np.matrix` by design, whereas with the current PR I'd have to review the two implementations to see if they match.

Reply:
Because `skip_fixed=False` is raising a `RuntimeError` :) But you are right, I will fix `matrix()` and then do as you suggest, makes much more sense. Sorry, I forgot this, but we talked about it offline.